docs/configure.asciidoc

```diff
@@ -0,0 +1,326 @@
+      "FileLogDirectory": "C:\\Logs" <1>
+    }
+  }
+}
+----
+<1> This example sets the file log directory to `C:\Logs` which enables diagnostic file logging.
+
+// Order of operations
+Configuration from the "Elastic:OpenTelemetry" section of the `IConfiguration` instance will be
+bound to the `ElasticOpenTelemetryOptions` instance used to configure the Elastic distribution.
+
+To learn more about the Microsoft configuration system, see
+https://learn.microsoft.com/en-us/aspnet/core/fundamentals/configuration[Configuration in ASP.NET Core].
+
+[discrete]
+[[configure-manual-configuration]]
+==== Manual configuration
+
+// What and why
+In all other scenarios, configuration can be achieved manually in code.
+
+// How
+This is done by creating an instance of `ElasticOpenTelemetryBuilderOptions` and passing it to the
+`ElasticOpenTelemetryBuilder` constructor or an overload of the `AddElasticOpenTelemetry` extension
+method on the `IServiceCollection`.
+
+// Example
+For example, in traditional console applications, you can configure the
+Elastic Distribution for OpenTelemetry .NET like this:
+
+[source,csharp]
+----
+using Elastic.OpenTelemetry;
+using Elastic.OpenTelemetry.Configuration;
+using Elastic.OpenTelemetry.Extensions;
+using Microsoft.Extensions.DependencyInjection;
+using OpenTelemetry;
+
+var services = new ServiceCollection();
+
+var builderOptions = new ElasticOpenTelemetryBuilderOptions <1>
+{
+	DistroOptions = new ElasticOpenTelemetryOptions <2>
+	{
+		FileLogDirectory = "C:\\Logs", <3>
+	}
+};
+
+await using var session = new ElasticOpenTelemetryBuilder(builderOptions) <4>
+	.WithTracing(b => b.AddSource("MySource"))
+	.Build();
+----
+<1> Create an instance of `ElasticOpenTelemetryBuilderOptions`
+<2> Create an instance of `ElasticOpenTelemetryOptions` and configure the file log directory by
+setting the corresponding property.
+<3> This example sets the file log directory to `C:\Logs` which enables diagnostic file logging.
+<4> Pass the `ElasticOpenTelemetryBuilderOptions` instance to the `ElasticOpenTelemetryBuilder` constructor
+to configure the Elastic Distribution for OpenTelemetry .NET.
+
+// Order of operations
+
+////
+What are the available configuration options?
+////
+[discrete]
+[[configure-configuration-options]]
+=== Configuration options
+
+////
+Is this true? Is the distro a wrapper around the OpenTelemetry .NET agent
+which is a wrapper around the general OpenTelemetry SDK? 🌀
+////
+Because the {distro_name} ("the distro") is a wrapper around the https://github.com/open-telemetry/opentelemetry-{language_lc}-instrumentation[OpenTelemetry {language} agent], it supports both:
+
+* General OpenTelemetry SDK configuration options
+* Elastic-specific configuration options that are only available when using the distro
+
+[discrete]
+[[configure-otel-sdk-options]]
+==== OpenTelemetry SDK configuration options
+
+////
+Is this true? Are there any options that aren't supported?
+Any options that shouldn't be used?
```

Comment:
No, we actively don't want to limit this. As of today we are not aware of any option that should not be used.